File: packages/hydra-java/src/main/haskell/Hydra/Sources/Java/Manifest.hs
-- | Package manifest for hydra-java.
--
-- Owns the Java coder DSL sources. See feature_290_packaging-plan.md,
-- "Sync system redesign / Package manifests".

module Hydra.Sources.Java.Manifest (
  mainModules,
  testModules,
  dslTypeModules,
) where

import Hydra.Kernel

import qualified Hydra.Sources.Java.Coder as JavaCoder
import qualified Hydra.Sources.Java.Environment as JavaEnvironmentSource
import qualified Hydra.Sources.Java.Language as JavaLanguageSource
import qualified Hydra.Sources.Java.Names as JavaNamesSource
import qualified Hydra.Sources.Java.Serde as JavaSerdeSource
import qualified Hydra.Sources.Java.Syntax as JavaSyntax
import qualified Hydra.Sources.Java.Testing as JavaTesting
import qualified Hydra.Sources.Java.Utils as JavaUtilsSource

mainModules :: [Module]
mainModules = [
  JavaCoder.module_,
  JavaEnvironmentSource.module_,
  JavaLanguageSource.module_,
  JavaNamesSource.module_,
  JavaSerdeSource.module_,
  JavaSyntax.module_,
  JavaTesting.module_,
  JavaUtilsSource.module_]

-- | Modules in this package whose type definitions should produce derived
-- DSL wrapper modules (Hydra/Dsl/Java/<Name>.hs). The DSL generator
-- consumes this list to emit constructors / accessors / withXxx updaters
-- for each TypeDefinition. Term-only modules (Coder, Serde, Language,
-- Names, Testing, Utils) are deliberately excluded. Extend the list when
-- a new type-defining module needs DSL wrappers.
dslTypeModules :: [Module]
dslTypeModules = [
  JavaEnvironmentSource.module_,
  JavaSyntax.module_]

testModules :: [Module]
testModules = []

File: packages/hydra-java/src/main/java/hydra/sources/java/Manifest.java
package hydra.sources.java;
import hydra.packaging.Module;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * Package manifest for hydra-java.
 *
 * Mirror of packages/hydra-java/src/main/haskell/Hydra/Sources/Java/Manifest.hs.
 * Owns the Java coder DSL sources.
 *
 * <p>Exposes {@link #mainModules} and {@link #testModules}, used by
 * build tooling to enumerate which Hydra modules belong to the
 * hydra-java package.</p>
 */
public class Manifest {
    public static final List<Module> mainModules = Arrays.asList(
        Coder.module_,
        Environment.module_,
        Language.module_,
        Names.module_,
        Serde.module_,
        Syntax.module_,
        Testing.module_,
        Utils.module_);

    public static final List<Module> testModules = Collections.emptyList();
}
